dataplane/linux/bpf_ep_mgr.go

```diff
@@ -288,6 +292,22 @@ func (m *bpfEndpointManager) OnUpdate(msg interface{}) {
 	}
 }
 
+func (m *bpfEndpointManager) onInterfaceAddrsUpdate(update *ifaceAddrsUpdate) {
+	if update.Addrs != nil && update.Addrs.Len() > 0 {
+		log.Debugf("Interface %+v received address update %+v", update.Name, update.Addrs)
+		update.Addrs.Iter(func(item interface{}) error {
+			ip := net.ParseIP(item.(string))
+			if ip.To4() != nil {
+				m.ifaceToIpMap[update.Name] = ip
```

Comment:
Can get spurious updates so it is worth avoiding marking the interface dirty if the IP hasn't changed